### PR TITLE
e2e/syncer: check deployment available independently and output yaml

### PR DESCRIPTION
**Pull request by sttts**
Towards https://github.com/kcp-dev/kcp/issues/947.

### CODE REVIEW

File: test/e2e/syncer/syncer_test.go [hunk 1]
```diff
@@ -153,14 +154,28 @@ func TestSyncerLifecycle(t *testing.T) {
...
 		}
 		return true
 	}, wait.ForeverTestTimeout, time.Millisecond*100, "downstream deployment %s/%s was not synced", downstreamNamespaceName, upstreamDeployment.Name)
 
+	t.Logf("Check for available replicas if downstream is capable of actually running the deployment")
+	expectedAvailableReplicas := int32(1)
+	require.Eventually(t, func() bool {
+		deployment, err = downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Errorf("saw an error waiting for downstream deployment %s/%s to be created: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
```
Comment (ncdc) [inline test/e2e/syncer/syncer_test.go:166]:
```suggestion
			t.Errorf("error getting downstream deployment %s/%s: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
```

Comment (ncdc) [reply test/e2e/syncer/syncer_test.go:166]: Also, need to `return false`?

File: test/e2e/syncer/syncer_test.go [hunk 2]
```diff
@@ -153,14 +154,28 @@ func TestSyncerLifecycle(t *testing.T) {
...
-		// Check for available replicas if downstream is capable of actually running the deployment
-		if len(framework.TestConfig.PClusterKubeconfig()) > 0 && expectedAvailableReplicas != deployment.Status.AvailableReplicas {
-			return false
 		}
 		return true
 	}, wait.ForeverTestTimeout, time.Millisecond*100, "downstream deployment %s/%s was not synced", downstreamNamespaceName, upstreamDeployment.Name)
 
+	t.Logf("Check for available replicas if downstream is capable of actually running the deployment")
+	expectedAvailableReplicas := int32(1)
+	require.Eventually(t, func() bool {
```
Comment (ncdc) [inline test/e2e/syncer/syncer_test.go:163]: Should we skip all of this if len(framework.TestConfig.PClusterKubeconfig()) is 0?

File: test/e2e/syncer/syncer_test.go (hunk 1 above)
Comment (shawn-hurley) [reply test/e2e/syncer/syncer_test.go:166]: Should it be a test error we can't get it if the `require.Eventually` is going to retry?

Comment (ncdc) [reply test/e2e/syncer/syncer_test.go:166]: We previously successfully retrieved it. Maybe this should just be `require.NoError(t, err)`?

File: test/e2e/syncer/syncer_test.go (hunk 2 above)
Comment (sttts) [reply test/e2e/syncer/syncer_test.go:163]: done

File: test/e2e/syncer/syncer_test.go (hunk 1 above)
Comment (sttts) [reply test/e2e/syncer/syncer_test.go:166]: done

**Review by davidfestal — APPROVED** [review]

**Pull request merged**